src/mito2/src/read/projection.rs

```diff
@@ -229,7 +227,7 @@ impl ProjectionMapper {
 #[derive(Debug, Clone, Copy)]
 enum BatchIndex {
     /// Index in primary keys.
```

Comment:
```suggestion
    /// Index and column id in primary keys.
```

Reply:
This PR is quite large, so I will split it into multiple smaller PRs. The issues raised here will be addressed in the subsequent PRs.